setup.py

```diff
@@ -235,27 +234,32 @@ def get_build_overrides():
     from numpy.distutils.command.build_clib import build_clib
     from numpy.distutils.command.build_ext import build_ext
 
-    def _is_using_gcc(obj):
-        is_gcc = False
+    def _is_using_old_gcc(obj):
+        is_old_gcc = False
         if obj.compiler.compiler_type == 'unix':
-            cc = sysconfig.get_config_var("CC")
-            if not cc:
-                cc = ""
-            compiler_name = os.path.basename(cc)
-            is_gcc = "gcc" in compiler_name
-        return is_gcc
+            cc = obj.compiler.compiler[0]
+            if cc == "gcc":
+                out = subprocess.run([cc, '-dumpversion'],
+                                     stdout=subprocess.PIPE,
+                                     stderr=subprocess.PIPE,
+                                    )
+                ver = float(out.stdout)
```

Comment:
this will fail for something like "5.4.0"

Reply:
fixed